rancher/service_prune.go

```diff
@@ -0,0 +1,45 @@
+package rancher
+
+import (
+	"github.com/rancher/rancher-compose-executor/project"
+)
+
+type RancherServicePrune struct {
+	ExistingServiceNames map[string]string
+	DeleteServiceIds     []string
+}
+
+func (r *RancherServicePrune) Prune(context *Context, project *project.Project) error {
+	if context.Stack == nil {
+		return nil
+	}
+
+	r.ExistingServiceNames = map[string]string{}
```

Comment:
Make the value here a `client.Service` so that `Client.Service.ById` doesn't need to be called twice for each service.